File: arrhash/longestseq.py
def longestseq(nums):
    first_array = []
    nums_set = set(nums)
    for n in nums:
        if (n-1 not in nums_set):
            first_array.append(n)
    max_length = 0
 

    for first in first_array:
        c = 1
        i = 1
        while first + i in nums_set:
            c += 1
            i += 1
        max_length = max(max_length, c)
            
    return max_length

nums = [9,1,4,7,3,-1,0,5,8,-1,6]
result = longestseq(nums)
print(result)

File: arrhash/longestseq.js
let nums = [9, 1, 4, 7, 3, -1, 0, 5, 8, -1, 6];

function longestseq(nums) {
  const firsts_array = [];
  const nums_set = new Set(nums);

  for (let n of nums) {
    if (!nums_set.has(n - 1)) {
      firsts_array.push(n);
    }
  }
  let max_length = 0;
  for (let first of firsts_array) {
    let c = 1;
    let i = 1;
    while (nums_set.has(first + i)) {
      c++;
      i++;
    }
    max_length = Math.max(max_length, c);
  }
  return max_length;
}

const result = longestseq(nums);
console.log(result);

document.getElementById("result").textContent = "Longest seq: " + result;
